### PR TITLE
Loading transaction examples

### PR DESCRIPTION
**Pull request by rizwansaeed**
Mainly need to review TransactionTests.java to check the TransactionRequest for totalConsideration, units and price for listed, otc and cash

### CODE REVIEW

File: sdk/src/test/java/com/finbourne/lusid/integration/TransactionsTests.java [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+        Portfolio portfolio = this.transactionPortfoliosApi.createPortfolio(scope, request);
+
+        assertEquals(portfolio.getId().getCode(), originalPortfolioId);
+
+        String portfolioId = portfolio.getId().getCode();
+
+        //  create the transaction
+        TransactionRequest transaction = new TransactionRequest()
+                .transactionId(UUID.randomUUID().toString())
+                .type("Buy")
```
Comment (steco) [inline sdk/src/test/java/com/finbourne/lusid/integration/TransactionsTests.java]: Probably best to use "FundsIn" transaction code, then you only need to specify Units not total consideration.  We should make TotalConsideration nullable, but in the meantime make it zero

File: sdk/src/test/java/com/finbourne/lusid/integration/TransactionsTests.java [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+                //  Cash instruments are identified using CCY_ followed by the ISO currency codes.
+                //  Cash instruments do not need to be created before use
+                .instrumentUid("CCY_GBP")
+                .totalConsideration(new CurrencyAndAmount().currency("GBP").amount(1000.0))
```
Comment (steco) [inline sdk/src/test/java/com/finbourne/lusid/integration/TransactionsTests.java]: Units, see above

File: sdk/src/test/java/com/finbourne/lusid/integration/TransactionsTests.java [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        String portfolioId = portfolio.getId().getCode();
+
+        //  create the transaction
+        TransactionRequest transaction = new TransactionRequest()
+                .transactionId(UUID.randomUUID().toString())
+                .type("Buy")
+
+                //  Cash instruments are identified using CCY_ followed by the ISO currency codes.
+                //  Cash instruments do not need to be created before use
+                .instrumentUid("CCY_GBP")
```
Comment (steco) [inline sdk/src/test/java/com/finbourne/lusid/integration/TransactionsTests.java]: For later, we need to decide if this is the right way to do this, and if so provide a utility to generate the CCY_GBP from just "GBP"

File: sdk/src/test/java/com/finbourne/lusid/integration/TransactionsTests.java [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+                .totalConsideration(new CurrencyAndAmount().currency("GBP").amount(1000.0))
+                .transactionDate(effectiveDate)
+                .settlementDate(effectiveDate)
+
+                //  Cash uploaded with a unit and price of 1
+                .units(1.0)
+                .transactionPrice(new TransactionPrice().price(1.0))
```
Comment (steco) [inline sdk/src/test/java/com/finbourne/lusid/integration/TransactionsTests.java]: no need to provide price

File: sdk/src/test/java/com/finbourne/lusid/integration/TransactionsTests.java [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+                .map(inst -> inst.getLusidInstrumentId())
+                .collect(Collectors.toList())
+                .get(0);
+
+        //  create the transaction
+        TransactionRequest transactionRequest = new TransactionRequest()
+                .transactionId(UUID.randomUUID().toString())
+                .type("Buy")
+                .instrumentUid(swapId)
+                .totalConsideration(new CurrencyAndAmount().currency("GBP").amount(0.0))
```
Comment (steco) [inline sdk/src/test/java/com/finbourne/lusid/integration/TransactionsTests.java]: Add a comment "Further discussion required on desired swap representation" or something similiar

File: sdk/src/test/java/com/finbourne/lusid/integration/TransactionsTests.java [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+        //  get the trade
+        VersionedResourceListOfTransaction transactions = this.transactionPortfoliosApi.getTransactions(scope,
+                portfolioId, null, null, null, null, null, null, null, null);
+
+        assertEquals(1, transactions.getValues().size());
+        assertEquals(transaction.getTransactionId(), transactions.getValues().get(0).getTransactionId());
+    }
+    
+    @Test
+    public void load_cash() throws ApiException {
```
Comment (steco) [inline sdk/src/test/java/com/finbourne/lusid/integration/TransactionsTests.java]: load_cash_transaction

File: sdk/src/test/java/com/finbourne/lusid/integration/TransactionsTests.java (hunk 2 above)
Comment (rizwansaeed) [reply sdk/src/test/java/com/finbourne/lusid/integration/TransactionsTests.java]: Unfortunately it looks like totalConsideration is mandatory

File: sdk/src/test/java/com/finbourne/lusid/integration/TransactionsTests.java (hunk 5 above)
Comment (rizwansaeed) [reply sdk/src/test/java/com/finbourne/lusid/integration/TransactionsTests.java]: raised https://finbourne.atlassian.net/browse/DEV-2183

**Pull request merged**